sdk/storage/azblob/blob/client.go

```diff
@@ -381,6 +383,168 @@ func (b *Client) download(ctx context.Context, writer io.WriterAt, o downloadOpt
 	return count, nil
 }
 
+// downloadFile downloads an Azure blob to a Writer. The blocks are downloaded parallely,
+// but written to file serially
+func (b *Client) downloadFile(ctx context.Context, writer io.Writer, o downloadOptions) (int64, error) {
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
This logic is already in place for azcopy and copytool. The user still has a way to force parallel writing using DownloadBuffer() method. This will also add new parameter to the options. Do we still need this?